src/app/use-cases/validate-document.ts

```diff
@@ -18,25 +18,51 @@ import {
   DocumentList,
   IsDocumentUploaded,
 } from "../../signature-request/document";
-import { dispatch } from "./status-signature-request";
 
 import { GetDocumentPayload } from "./get-document";
+import { dispatchOnDocument, DocumentAction } from "./status-document";
 
+const documentNotFoundError = new EntityNotFoundError("Document not found");
 export const addUrlToDocument =
   (documentId: Document["id"], url: string) => (request: SignatureRequest) =>
     pipe(
       request.documents,
-      findIndex((document) => document.id === documentId),
+      A.findIndex((document) => document.id === documentId),
       O.chain((index) =>
         pipe(
           request.documents,
-          modifyAt(index, (document) => ({
+          A.modifyAt(index, (document) => ({
             ...document,
             url,
           }))
         )
       ),
-      E.fromOption(() => new EntityNotFoundError("Document not found")),
+      E.fromOption(() => documentNotFoundError),
+      E.chainW(validate(DocumentList, "Unable to validate document list"))
```

Comment:
why do you want to revalidate request.documents which is already a document list?

Reply:
If it is not revalidated it fails to verify the branded types. 
`Property '[_brand]' is missing in type`